alleycats-core/src/main/scala/alleycats/std/map.scala

```diff
@@ -45,11 +45,24 @@ trait MapInstances {
             G.map(f(a))((k, _))
           }) { chain => chain.foldLeft(Map.empty[K, B]) { case (m, (k, b)) => m.updated(k, b) } }
 
+      override def mapAccumulate[S, A, B](init: S, fa: Map[K, A])(f: (S, A) => (S, B)): (S, Map[K, B]) = {
+        val iter = fa.iterator
+        var s = init
+        val m = Map.newBuilder[K, B]
```

Comment:
What do you think about adding the call of `m.sizeHint`? I haven't a strong opinion it's required, considering extra computing of `Map` length, but someone used it in cats.

Reply:
good call. I'll update that.

Reply:
done.